core/trino-main/src/main/java/io/trino/server/protocol/JsonEncodingUtils.java

```diff
@@ -427,5 +443,20 @@ public void encode(JsonGenerator generator, ConnectorSession session, Block bloc
                 default -> generator.writePOJO(value);
             }
         }
+
+        private Object roundParametricTypes(Object value)
```

Comment:
can this PR be more iterative. It is hard to follow what is happening in this commit.

Reply:
I've tried to do just that and I failed. I can walk you through it offline